lib/msal-common/src/utils/RequestUtils.ts

```diff
@@ -0,0 +1,239 @@
+     * @param codeChallengeMethod
+     */
+    static addCodeChallengeParams(
+        params: Map<string, string>,
+        codeChallenge: string,
+        codeChallengeMethod: string
+    ): void {
+        if (codeChallenge && codeChallengeMethod) {
+            params.set(
+                `${AADServerParamKeys.CODE_CHALLENGE}`,
+                encodeURIComponent(codeChallenge)
+            );
+            params.set(
+                `${AADServerParamKeys.CODE_CHALLENGE_METHOD}`,
+                encodeURIComponent(codeChallengeMethod)
+            );
+        } else {
+            throw ClientConfigurationError.createInvalidCodeChallengeParams();
+        }
+    }
+
+    /**
+     * add the `authorization_code` passed by the user to exchange for a token
+     * @param params
+     * @param code
+     */
+    static addAuthorizationCode(params: Map<string, string>, code: string): void {
+        params.set(`${AADServerParamKeys.CODE}`, encodeURIComponent(code));
+    }
+
+    /**
+     * add the `authorization_code` passed by the user to exchange for a token
+     * @param params
+     * @param code
+     */
+    static addDeviceCode(params: Map<string, string>, code: string): void {
+        params.set(`${AADServerParamKeys.DEVICE_CODE}`, encodeURIComponent(code));
+    }
+
+    /**
+     * add the `code_verifier` passed by the user to exchange for a token
+     * @param params
+     * @param codeVerifier
+     */
+    static addCodeVerifier(params: Map<string, string>, codeVerifier: string): void {
+        params.set(`${AADServerParamKeys.CODE_VERIFIER}`, codeVerifier);
+    }
+
+    /**
+     * add client_secret
+     * @param params
+     * @param clientSecret
+     */
+    // TODO uncomment when confidential client flow is added.
+    // static addClientSecret(params: Map<string, string>, clientSecret: string): void {
+    //     params.set(`${AADServerParamKeys.CLIENT_SECRET}`, clientSecret);
+    // }
+
+    static addGrantType(params: Map<string, string>, grantType: string): void {
+        params.set(`${AADServerParamKeys.GRANT_TYPE}`, encodeURIComponent(grantType));
+    }
+
+    /**
+     * add "ContentType:" header
+     * @param headers
+     */
+    static addContentTypeHeader(headers: Map<string, string>): void {
+        headers.set(HeaderNames.CONTENT_TYPE, Constants.URL_FORM_CONTENT_TYPE);
+    }
+
+    /**
+     * addLibraryData
+     * @param headers
+     */
+    static addLibrarydataHeaders(headers: Map<string, string>): void {
+        // library version
+        headers.set(
+            `${AADServerParamKeys.X_CLIENT_SKU}`,
+            encodeURIComponent(Constants.LIBRARY_NAME)
+        );
+        headers.set(
+            `${AADServerParamKeys.X_CLIENT_VER}`,
+            encodeURIComponent("0.0.1")
+        );
+    }
+
+    /**
+     * Utility to create a URL from the params map
+     * @param paramsMap
+     * @param authority
+     */
+    static createQueryString(paramsMap: Map<string, string>): string {
```

Comment:
Similar to the comment on `RequestValidator`, I think every function in this file probably belongs somewhere in the `request` folder else except for this one as `createQueryString`